### ISSUE TITLE
v4 dropdown-header should be focusable for accesibility

### ISSUE THREAD

**Issue by tmorehouse** [twbs/bootstrap#23336]
In order to allow screen readers to read out dropdown headings, they should be focusable (but not actionalable), as dropdown headers help convey meaning to the drop-down items below them.

Such as this situation:
![image](https://user-images.githubusercontent.com/2781561/29225777-f85d8602-7ea5-11e7-8e4f-fc6f08496f1b.png)

Without being able to "read" the dropdown-header, the user wouldn't know which context the "user" dropdown-item's are for.

The current alternative workaround is to add `.sr-only` text to the items that replicate the header text in each `dropdown-item`.

**Comment by patrickhlauke** [comment]
Gut feeling is that they should not be focusable, but rather that there needs to be a programmatic relationship between the headers and their respective links/options. Simplest (but most verbose) thing I can think of off the top of my head is an `aria-describedby` attribute on all the items pointing back to their respective header.

**Comment by tmorehouse** [comment]
Yeah, that would be another option (`aria-describedby`)

**Comment by tmorehouse** [comment]
Maybe just mentioning it in the docs would be a good enough "fix"...

**Comment by patrickhlauke** [comment]
docs should also have a good example of the solution in action. but yes, this will boil down to advice prose as well. will do some experimenting/testing with AT.

**Comment by tmorehouse** [comment]
I found another markup that works well for dropdown headers and ARIA semantic markup that doesn't need aria-describedby:

```html
<div class="dropdown-menu">
  <div role="group">
    <h5 class="dropdown-header">Header 1</h5>
    <a href="#" class="dropdown-item">Action One</a>
    <a href="#" class="dropdown-item">Action Two</a>
  </div>
  <div role="group">
    <h5 class="dropdown-header">Header 2</h5>
    <a href="#" class="dropdown-item">Action One</a>
    <a href="#" class="dropdown-item">Action Two</a>
  </div>
</html>
```

By nature of above the markup, the headers automatically get associated with their respective dropdown items.

**Comment by mdo** [comment]
Are these any changes we need to make here or is this safe to close?

**Comment by mdo** [comment]
Closing as stale.

I confirmed that the above example does indeed still work as intended with at least the styling of dropdown menu items. Unsure if we need this added as an example in the docs, but if so, please feel free to open a PR for `main` that we could backport into `v4-dev`. Otherwise, I think this is safe to close.

<3
